socorro/unittest/cron/jobs/test_monitoring.py

```diff
@@ -0,0 +1,256 @@
+import json
+from contextlib import contextmanager, nested
+
+import pytest
+import mock
+from configman.dotdict import DotDict
+
+from socorro.cron.jobs.monitoring import (
+    DependencySecurityCheckCronApp,
+    DependencySecurityCheckFailed,
+    Vulnerability,
+)
+from socorro.unittest.cron.jobs.base import IntegrationTestBase
+
+
+# Ideally this'd be a pytest yielding fixture, but fixtures aren't easily used
+# with unittest.TestCase subclasses.
+@contextmanager
+def mock_popen(return_code, output='', error_output=''):
+    popen_patch = mock.patch('socorro.cron.jobs.monitoring.Popen')
+    popen = popen_patch.start()
+
+    process = popen.return_value
+    process.returncode = return_code
+    process.communicate.return_value = (output, error_output)
+
+    yield popen
+
+    popen_patch.stop()
```

Comment:
When I was working on Antenna, I made a bunch of things like this as pytest fixtures. This is fine, but having it as a pytest fixture might be slightly more ergonomic.

https://github.com/mozilla-services/antenna/blob/1020bb2883ae9ab8d654a8a4e2905b3da68e6a45/tests/unittest/conftest.py#L117

Reply:
Though, maybe switching to a pytest fixture doesn't help a whole lot here. I wonder if we mock popen elsewhere in Socorro.

Reply:
The problem is that pytest fixtures don't work right with `unittest.TestCase` subclasses. You can't do the "include an argument named after the fixture" magic, and instead have to mark the test functions as manually using the fixture. And I couldn't even get that to work, actually; the docs only mention using class-scoped fixtures in this way.

However, the only reason I was using `IntegrationTestBase` was for `_setup_config_manager`, which I'm not using anymore. I'll refactor the tests to make this a proper fixture,

Reply:
Oh right. I totally forgot about that. I have vague memories of that issue being alleviated with pytest 4. I should look that up, but for now I'm just going to hope it's true and feel good about the world.